autogen/agentchat/conversable_agent.py

```diff
@@ -1184,6 +1184,18 @@ def register_function(self, function_map: Dict[str, Callable]):
         """
         self._function_map.update(function_map)
 
+    def define_function(self, signature: Dict):
```

Comment:
Why do we need this function? Isn't it a "add_function" rather than "define_function"? If so, we should rename it.

Reply:
Can we use the existing "register_function" instead?

Reply:
Resolved. I finally use `update_function_signature`. It is more appropriate.